util/fscrypt_completion

```diff
@@ -0,0 +1,225 @@
+# bash completion for fscrypt                              -*- shell-script -*-
+
+# Output list of possible mount points
+_fscrypt_mountpoints()
+{
+    fscrypt status 2>/dev/null | \
+        command sed \
+            -e '/^\//!d' \
+            -e '/Yes$/!d' \
+            -e 's/ .*//'
+}
+
+# Output list of possible policy or protector IDs
+# $1: the mount point on which policies are looked for.
+# $2: the section (policy or protector) to retrieve
+_fscrypt_status_section()
+{
+    fscrypt status "$1" 2>/dev/null | \
+        command sed \
+            -e "/^${2^^} /",'/^$/!d' \
+            -e "s/${2^^}//" \
+            -e "s/\s.*//"
+}
+
+_fscrypt()
+{
+    local cur prev words cword split
+    local GLOBAL_OPTS="--verbose --quiet --help"
+    _init_completion -s -n : || return
+
+    case $prev in
+        --drop-caches)
+            COMPREPLY=($(compgen -W "true false" -- "${cur}"))
+            return ;;
+        --key)
+            _filedir
+            return ;;
+        --name)
+            # New value, nothing to complete
+            return ;;
+        --policy|--protector|--unlock-with)
+            if [[ $cur = *:* ]]; then
+                # Complete with IDs of the given mountpoint
+                local section="${prev#--}"
+                [[ $section = unlock-with ]] && section=protector
+                local mountpoint="${cur%:*}" id="${cur#*:}"
+                COMPREPLY=($(compgen -W "$(_fscrypt_status_section \
+                                           "${mountpoint}" "${section}")" \
+                             -- "${id}"))
+            else
+                # Complete with mountpoints, with colon and without ending space
+                COMPREPLY=($(compgen -W "$(_fscrypt_mountpoints)" \
+                             -- "${cur}" | sed s/\$/:/))
+                compopt -o nospace
+            fi
+            return ;;
+        --source)
+            COMPREPLY=($(compgen -W "pam_passphrase custom_passphrase raw_key" \
+                         -- "${cur}"))
+            return ;;
+        --time)
+            # It's a time, hard to complete a number…
+            return ;;
+        --user)
+            COMPREPLY=($(compgen -u -- "${cur}"))
+            return ;;
+    esac
+
+    local -a positional
+    local cword cmd
```

Comment:
The `cmd` variable is unused.

BTW, I found this by running `shellcheck`.  Can you please run `shellcheck` and fix any warnings it makes sense to fix?  At the moment all the other shell scripts in this project are `shellcheck`-clean, and `make lint` enforces this.  If there are some specific conventions for bash completion scripts that don't pass `shellcheck`, we can make an exception, but it would be preferable not to.

Reply:
Well catched.  I checked the file with the scop/bash-completion test suite, but I didn't use shellcheck.

I fixed most of them, and add a comment for all explicitly disabled warnings.  The one I cannot resolve is SC2016 on your awk expressions: `$` is wrongly recognized by shellcheck as a bash variable.  Do you have an idea on the best way to resolve it?

Should I add something in the Makefile for that file to be checked via `make lint`?

Reply:
Yes, if possible shellcheck should be run by `make lint`.